connect_ext_test.go

```diff
@@ -2079,6 +2080,108 @@ func TestHandlerReturnsNilResponse(t *testing.T) {
 	assert.Equal(t, panics, 2)
 }
 
+func TestStreamUnexpectedEOF(t *testing.T) {
+	t.Parallel()
+	testcases := map[string]http.HandlerFunc{
+		"stream_unexpected_eof": func(responseWriter http.ResponseWriter, request *http.Request) {
+			_, _ = io.Copy(io.Discard, request.Body)
```

Comment:
Thanks for adding additional tests!
Sorry, but just one last thought: this could be de-duplicated a bit by moving the first four lines of each handler into the `HandleFunc` call before calling `testcase`. Then, since `payload` and `head` never actually change, it should also be safe to move them above. Then it'd look a bit more like this:

```go
	payload := []byte(`{"number": 42}`)
	head := [5]byte{}
	binary.BigEndian.PutUint32(head[1:], uint32(len(payload)))

	testcases := map[string]http.HandlerFunc{
		"stream_unexpected_eof": func(responseWriter http.ResponseWriter, request *http.Request) {
			_, _ = responseWriter.Write(head[:])
			_, _ = responseWriter.Write(payload)
		},
		"stream_partial_payload": func(responseWriter http.ResponseWriter, request *http.Request) {
			_, _ = responseWriter.Write(head[:])
			_, _ = responseWriter.Write(payload[:len(payload)-1])
		},
		"stream_partial_frame": func(responseWriter http.ResponseWriter, request *http.Request) {
			_, _ = responseWriter.Write(head[:4])
		},
	}
```

WDYT?

Reply:
Could group the handler and test logic together?
```go
	testcases := []struct {
		name    string
		handler http.HandlerFunc
		runner  func(t *testing.T)
	}{{
		name: "stream_unexpected_eof",
		handler: func(responseWriter http.ResponseWriter, request *http.Request) {
			_, _ = responseWriter.Write(head[:])
			_, _ = responseWriter.Write(payload)
		},
		runner: func(t *testing.T) {
			const upTo = 2
			request := connect.NewRequest(&pingv1.CountUpRequest{Number: upTo})
			request.Header().Set("Test-Case", t.Name())

			stream, err := client.CountUp(context.Background(), request)
			assert.Nil(t, err)
			for stream.Receive() {
				assert.Equal(t, stream.Msg().Number, 42)
			}
			assert.NotNil(t, stream.Err())
			assert.Equal(t, connect.CodeOf(stream.Err()), connect.CodeUnknown)
			assert.True(t, errors.Is(stream.Err(), io.ErrUnexpectedEOF))
		},
	}}
```